src/Interpreters/GinFilter.cpp

```diff
@@ -16,20 +16,12 @@
 namespace DB
 {
 
-namespace ErrorCodes
-{
-    extern const int BAD_ARGUMENTS;
-}
-
-GinFilterParameters::GinFilterParameters(size_t ngrams_, UInt64 max_rows_per_postings_list_)
-    : ngrams(ngrams_)
+GinFilterParameters::GinFilterParameters(String tokenizer_, UInt64 max_rows_per_postings_list_)
```

Comment:
Looks like there was some special case for `UNLIMITED_ROWS_PER_POSTINGS_LIST` (l. 23) but it can no longer be reached. I guess we should allow this case?

Reply:
yes, we should still allow that. I fixed it